app/models/project.rb

```diff
@@ -146,4 +146,10 @@ def self.public_projects
                 .group("project_id")
                 .pluck(:project_id))
   end
+
+  def destroy
+    samples = Sample.find(sample_ids)
+    samples.each(&:destroy)
```

Comment:
Why not `.where(...).destroy_all`?

Reply:
mostly because i was unaware of that way of doing it. i think they're equivalent though, no?